platform/broadcom/sonic-platform-modules-dell/s5296f/sonic_platform/psu.py

```diff
@@ -0,0 +1,317 @@
+        if not is_valid:
+            return None
+
+        return float(input_power)
+
+    def get_powergood_status(self):
+        """
+        Retrieves the powergood status of PSU
+
+        Returns:
+            A boolean, True if PSU has stablized its output voltages and
+            passed all its internal self-tests, False if not.
+        """
+        status = False
+        is_valid, state = self.state_sensor.get_reading()
+        if is_valid:
+            if (state == 0x01):
+                status = True
+
+        return status
+
+    def get_mfr_id(self):
+        """
+        Retrives the Manufacturer Id of PSU
+
+        Returns:
+            A string, the manunfacturer id.
+        """
+        return self.fru.get_board_mfr_id()
+
+    def get_type(self):
+        """
+        Retrives the Power Type of PSU
+
+        Returns :
+            A string, PSU power type
+        """
+        board_product = self.fru.get_board_product()
+        if board_product is not None :
+            info =  board_product.split(',')
+            if 'AC' in info : return 'AC'
+            if 'DC' in info : return 'DC'
+        return None
+
+    def get_position_in_parent(self):
+        """
+        Retrieves 1-based relative physical position in parent device.
+        Returns:
+            integer: The 1-based relative physical position in parent
+            device or -1 if cannot determine the position
+        """
+        return self.index
+
+    def get_temperature_high_threshold(self):
+        """
+        Returns the high temperature threshold for PSU in Celsius
+        """
+        is_valid, high_threshold = self.temp_sensor.get_threshold("UpperCritical")
+        if not is_valid:
+            high_threshold = 105
+        high_threshold = "{:.2f}".format(high_threshold)
+
+        return float(high_threshold)
+
+    def get_voltage_low_threshold(self):
+        """
+        Retrieves the low threshold PSU voltage output
+        Returns:
+            A float number, the low threshold output voltage in volts,
+            e.g. 12.1
+        """
+        return 0.0
+
+    def get_voltage_high_threshold(self):
+        """
+        Returns PSU high threshold in Volts
+        """
+        return 0.0
+
+    def get_maximum_supplied_power(self):
+        """
+        Retrieves the maximum supplied power by PSU
+        Returns:
+            A float number, the maximum power output in Watts.
+            e.g. 1200.1
+        """
+        return float(750)
+
+    def set_status_led(self, color):
+        """
+        Sets the state of the PSU status LED
+        Args:
+            color: A string representing the color with which to set the PSU status LED
+                   Note: Only support green and off
+        Returns:
+            bool: True if status LED state is set successfully, False if not
+        """
+        # Hardware not supported
+        return False
```

Comment:
this will break our nightly tests.  Need to modify test scripts to skip PSU led tests on this platform.

Reply:
In platform.json, we have defined the PSU's status led can't be controlled. Due to which the platform_tests/api/test_psu_fans.py test case will be skipped.